src/BuildScriptGenerator/Node/NodePlatform.cs

```diff
@@ -181,7 +230,9 @@ public PlatformDetectorResult Detect(RepositoryContext context)
             var productionOnlyPackageInstallCommand = string.Format(
                 NodeConstants.ProductionOnlyPackageInstallCommandTemplate, packageInstallCommand);
 
-            if (string.IsNullOrEmpty(_nodeScriptGeneratorOptions.CustomRunBuildCommand))
+            if (string.IsNullOrEmpty(_nodeScriptGeneratorOptions.CustomRunBuildCommand) &&
```

Comment:
Here are priorities of which build command we should generate in the build script: `customRunBuildCommand ` > lerna 
 or lage commands > build/build:azure under script in `package.json` commands.  So I modifies some logics here hope it makes sense to you @kichalla @arroyc.